crates/hir-def/src/lib.rs

```diff
@@ -516,6 +516,7 @@ pub type LocalModuleId = Idx<nameres::ModuleData>;
 pub struct FieldId {
     pub parent: VariantId,
     pub local_id: LocalFieldId,
+    pub has_default: bool,
```

Comment:
I just realized we have t his in `FieldData` already, so we can't we reuse that instead? If we have the ID we can trivially index into the `variant_data` where required.